cfpb/consumerfinance.gov
# Add Elasticsearch DSL to support Elasticsearch 7 for ask_cfpb

consumerfinance.gov supports elasticsearch 2.4.1 for ccdb5_api and legacy haystack apps


## Additions

- consumerfinance.gov supports elasticsearch7 for ask_cfpb application


## Changes

- elasticsearch versions 2.4 and 7.9 are supported


## Modifications

- boolean feature flag added to switch between elasticsearch2 and elasticsearch7 code 

## How to test this PR

1. Confirm versions of `elasticsearch` are supported using
```
$ pip list|grep elastic
django-elasticsearch-dsl 7.1.4
elasticsearch            2.4.1
elasticsearch-dsl        7.2.2
elasticsearch7           7.9.1
```

2. Enable `ELASTICSEARCH_DSL_ASK` flag

3. Verify the ask-cfpb index runs, and elasticsearch7 is working
`manage.py update_index ask_cfpb`

4. Confirm the haystack indexing runs as before using elasticsearch2:
`cfgov/manage.py  search_index --rebuild -f --parallel`

5. Search Ask


## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
